command/agent/job_endpoint.go

```diff
@@ -141,12 +141,17 @@ func (s *HTTPServer) jobPlan(resp http.ResponseWriter, req *http.Request,
 		return nil, CodedError(400, "Job ID does not match")
 	}
 
-	// Http region takes precedence over hcl region
+	// Region in http request query param takes precedence over region in job hcl config
 	if args.WriteRequest.Region != "" {
 		args.Job.Region = helper.StringToPtr(args.WriteRequest.Region)
 	}
+	// If 'global' region is specified or if no region is given,
+	// default to region of the node you're submitting to
+	if args.Job.Region == helper.StringToPtr("global") ||
+		args.Job.Region == helper.StringToPtr("") || args.Job.Region == nil {
+		args.Job.Region = &s.agent.config.Region
+	}
```

Comment:
Cleaner syntax using pointer deferencing, and relies on boolean short circuiting: 

```
if args.Job.Region == nil || *args.Job.Region == "global" || *args.Job.Region == "" { 
..
```

Reply:
Done